spec/unit/type/k5login_spec.rb

```diff
@@ -131,6 +152,46 @@ def resource(attrs = {})
           end
         end
       end
+
+      context "#stat" do
+        let(:file) { described_class.new(:path => path) }
+        it "should return nil if the file does not exist" do
+          file[:path] = make_absolute('/foo/bar/baz/non-existent')
+
+          expect(file.stat).to be_nil
+        end
+
+        it "should return nil if the file cannot be stat'ed" do
+          dir = tmpfile('link_test_dir')
+          child = File.join(dir, 'some_file')
+          Dir.mkdir(dir)
+          File.chmod(0, dir)
+
+          file[:path] = child
+
+          expect(file.stat).to be_nil
+
+          # chmod it back so we can clean it up
+          File.chmod(0777, dir)
```

Comment:
do we want want a begin..ensure around this, otherwise if the test fails for some reason end you end up leaving a directory that is mode 0000 laying around.